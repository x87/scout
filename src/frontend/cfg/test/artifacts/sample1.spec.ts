import loop3 from 'samples/loop3';
import loop4 from 'samples/loop4';
import { decompile } from '../util';

[loop3, loop4].forEach((sample, i) =>
  describe(`sample input ${i + 1}`, () => {
    it(`has redundant 'continue' statements`, async () => {
      expect(await decompile(sample)).toMatchSnapshot();
    });
  })
);
