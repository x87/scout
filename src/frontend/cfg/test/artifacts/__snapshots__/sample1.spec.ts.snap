// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sample input 1 has redundant 'continue' statements 1`] = `
"
:NONAME_0
    NOP
    
    while IS_INT_LVAR_GREATER_THAN_NUMBER 0 100
        if 
            IS_INT_LVAR_GREATER_THAN_NUMBER 0 2
        then
            WAIT 1
            continue
        else
            WAIT 2
        end
    end
    
    
    repeat
        WAIT 3
        if 
            IS_INT_LVAR_GREATER_THAN_NUMBER 0 0
        then
            WAIT 1
        end
    until IS_INT_LVAR_GREATER_THAN_NUMBER 0 1
    
    
    while IS_INT_LVAR_GREATER_THAN_NUMBER 0 1
        WAIT 5
        continue
    end
    
    TERMINATE_THIS_SCRIPT
"
`;

exports[`sample input 2 has redundant 'continue' statements 1`] = `
"
:NONAME_0
    NOP
    
    repeat
        WAIT 0
        if 
            NOT IS_INT_LVAR_GREATER_OR_EQUAL_TO_NUMBER 0 0
        then
            WAIT 444
            GOTO -66
        end
        if 
            NOT IS_INT_LVAR_GREATER_OR_EQUAL_TO_NUMBER 0 0
        then
            WAIT 555
            continue
        else
            WAIT 1
        end
    until IS_INT_LVAR_GREATER_THAN_NUMBER 0 1
    
    TERMINATE_THIS_CUSTOM_SCRIPT
"
`;
